src/wallet/wallet.cpp

```diff
@@ -2804,6 +2812,90 @@ CRecipient CWallet::CreateLelantusMintRecipient(
     }
 }
 
+std::list<std::pair<spark::Coin, CSparkMintMeta>> CWallet::GetAvailableSparkCoins(const CCoinControl *coinControl) const {
+    EnsureSparkWalletAvailable();
+
+    LOCK2(cs_main, cs_wallet);
+    CWalletDB walletdb(strWalletFile);
+    std::list<std::pair<spark::Coin, CSparkMintMeta>> coins;
+    // get all unsued coins from spark wallet
+    std::vector<CSparkMintMeta> vecMints = sparkWallet->ListSparkMints(true, true);
+    for (const auto& mint : vecMints) {
+        if(mint.v == 0) // ignore 0 mints which where created to increase privacy
+            continue;
+        // get actual spark coin
+        spark::Coin coin = sparkWallet->getCoinFromMeta(mint);
+        coins.push_back(std::make_pair(coin, mint));
+    }
+
+    std::set<COutPoint> lockedCoins = setLockedCoins;
+
+    // Filter out coins that have not been selected from CoinControl should that be used
+    coins.remove_if([lockedCoins, coinControl](const std::pair<spark::Coin, CSparkMintMeta>& coin) {
+        COutPoint outPoint;
+
+        // ignore if the coin is not actually on chain
+        if (!spark::GetOutPoint(outPoint, coin.first)) {
```

Comment:
If this is like the equivalent Lelantus call, it will take a long time and make this call require a disk read for every Spark coin in our wallet.